tests/trainer/test_trainer.py

```diff
@@ -774,6 +774,36 @@ def predict_step(self, batch, *_):
             trainer_fn(model, ckpt_path="best")
 
 
+@pytest.mark.parametrize("fn", ("validate", "test", "predict"))
+def test_best_ckpt_evaluate_raises_error_with_multiple_ckpt_callbacks(tmpdir, fn):
+    """Test that an error is raised if best ckpt callback is used for evaluation configured with multiple
+    checkpoints."""
+
+    class TestModel(BoringModel):
+        def validation_step(self, batch, batch_idx):
+            self.log("foo", batch_idx)
+            self.log("bar", batch_idx + 1)
+            return super().validation_step(batch, batch_idx)
+
+    ckpt_callbacks = [ModelCheckpoint(monitor="foo", save_top_k=1), ModelCheckpoint(monitor="bar", save_top_k=1)]
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        limit_train_batches=1,
+        callbacks=ckpt_callbacks,
+        limit_test_batches=1,
+        limit_val_batches=1,
+        limit_predict_batches=1,
+    )
+
+    model = TestModel()
+    trainer.fit(model)
+
+    trainer_fn = getattr(trainer, fn)
+    with pytest.warns(UserWarning, match="best checkpoint path from first checkpoint callback"):
```

Comment:
Can this test be a unit-test calling `__set_ckpt_path`?

This warning appears before training, we don't need to do it...

Reply:
done!